### ISSUE TITLE
dataproc-client should just copy configuration from target cluster

### ISSUE THREAD

**Issue by karth295** [jasonmar/dataproc-client#6]
Dataproc calculates configuration for a cluster based on the cluster size and more importantly, the machine type used. Spark/Hadoop/Hive jobs inherit /etc/*/conf/ configs from where the driver is run, so jobs submitted from a dataproc-client node inherit all of their configuration from the client (originally a single node cluster!).

Single node clusters must house both the master daemons (NN/RM) and worker daemons (DN/NN), and also run MR/Spark jobs. To make space for all the overhead, we configure less memory for map and reduce tasks on single node clusters than standard clusters of the same machine type.

In addition, there are dramatically different configurations between standard, highmem, and highcpu machine types.

So tl;dr config properties on dataproc-client are incorrect today.

Instead, they should be copied from the master node of the target cluster, after the target cluster has finished booting up and running initialization actions. I would suggest copying the `/etc/*/conf/` directories from the master node.

**Comment by karth295** [comment]
Copying configs from the cluster also avoids issues with running `sed` such as https://github.com/jasonmar/dataproc-client/issues/5

**Comment by karth295** [comment]
One thing this does not solve: the client and cluster can have different versions of software if they were created from different Dataproc image versions. For example, the cluster can have Spark 2.4, but the client can have Spark 2.3, and that will cause errors. 

I even ran into differences between subminor versions -- the client had Tez 0.9.1, and the cluster had Tez 0.9.2. Dataproc's Tez configuration expects Tez libraries at /usr/lib/tez/tez-*-<version>.jar, and the 0.9.1 libraries were not found on the cluster, so the job failed to even start.

I will open a separate issue(s) for that.

**Comment by jasonmar** [comment]
This seems pretty easy to get out of sync, maybe we should include a way to quickly compare versions
